distributed/cli/tests/test_dask_worker.py

```diff
@@ -149,24 +167,55 @@ def test_no_nanny(loop):
 
 @pytest.mark.slow
 @pytest.mark.parametrize("nanny", ["--nanny", "--no-nanny"])
-def test_no_reconnect(nanny, loop):
-    with popen(["dask-scheduler", "--no-dashboard"]) as sched:
-        wait_for_port(("127.0.0.1", 8786))
+@pytest.mark.asyncio
+async def test_no_reconnect(nanny):
+    async with Scheduler(dashboard_address=":0") as s, Client(
+        s.address, asynchronous=True
+    ) as c:
         with popen(
             [
                 "dask-worker",
-                "tcp://127.0.0.1:8786",
+                s.address,
                 "--no-reconnect",
                 nanny,
                 "--no-dashboard",
             ]
         ) as worker:
-            sleep(2)
-            terminate_process(sched)
-        start = time()
-        while worker.poll() is None:
-            sleep(0.1)
-            assert time() < start + 30
+            await c.wait_for_workers(1)
+
+            (comm,) = s.stream_comms.values()
+            comm.abort()
+
+            await to_thread(worker.communicate, timeout=30)
```

Comment:
IMHO, the docs on [communicate](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate) are a bit fuzzy about what happens if the process terminates. Do we actually have to wait for `timeout` to pass or will it return once the process is dead?

Reply:
sounds good